calyx-py/test/correctness/matrix.py

```diff
@@ -0,0 +1,71 @@
+import calyx.builder as cb
+
+def insert_matmul_component(prog):
+    matmul = prog.component("main")
+
+    A = matmul.comb_mem_d2("A", 32, 4, 4, 3, 3, is_external=True)
+    B = matmul.seq_mem_d2("B", 32, 4, 4, 3, 3, is_external=True)
+    C = matmul.seq_mem_d2("C", 32, 4, 4, 3, 3, is_external=True)
+
+    mult = matmul.mult_pipe(32)
+    add = matmul.add(32)
+
+    acc = matmul.reg(32)
+
+    # matrix entries
+    a = matmul.reg(32)
+    b = matmul.reg(32)
+
+    # iterators
+    i = matmul.reg(3)
+    j = matmul.reg(3)
+    k = matmul.reg(3)
+
+    zero_acc = matmul.reg_store(acc, 0)
+    zero_i = matmul.reg_store(i, 0)
+    zero_j = matmul.reg_store(j, 0)
+    zero_k = matmul.reg_store(k, 0)
+
+    cond_i = matmul.lt_use(i.out, 4)
+    cond_j = matmul.lt_use(j.out, 4)
+    cond_k = matmul.lt_use(k.out, 4)
+
+    read_A = matmul.mem_load_d2(A, i.out, k.out, a, "read_A")
+    read_B = matmul.mem_load_d2(B, k.out, j.out, b, "read_B")
+    
+    write = matmul.mem_store_d2(C, i.out, j.out, acc.out, "write")
```

Comment:
Consider writing pseudocode comments (`i := 0`, `a := A[i][j]`, etc.) for some of these lines. The builder library's higher-level utilities are sure to confuse regular Calyx users, and I hope to eventually point to this Python file from the docs. It would be useful if this file were quite user-friendly!

Reply:
Definitely agree. I've add these changes in the latest commit

Reply:
Looking good! FYI you can dismiss a suggestion by hitting the "resolve conversation" button to the bottom-left!